app/models/privacy_notification.rb

```diff
@@ -0,0 +1,16 @@
+class PrivacyNotification < ActiveRecord::Base
+  PETITION_LIMIT = 5
+
+  has_one :signature, foreign_key: :uuid
+  has_many :signatures, -> { validated }, foreign_key: :uuid
+
+  has_many :petitions, lambda {
+    not_anonymized.moderated.distinct.by_most_recent.limit(PETITION_LIMIT)
```

Comment:
Maybe a class method on `Petition` might be in order here? e.g.

``` ruby
def for_privacy_notification(size = 5)
  not_anonymized.moderated.distinct.by_most_recent.limit(size)
end
```

Then the association becomes a lot cleaner:

``` ruby
has_many :petitions, -> { for_privacy_notification }, through: :signatures
```

Alternatively, if the above is polluting the `Petition` class too much for you then just move the lambda to a constant:

```
PETITION_LIMIT = 5
PETITION_SCOPE = lambda {
  not_anonymized.moderated.distinct.by_most_recent.limit(PETITION_LIMIT)
}

has_many :petitions, PETITION_SCOPE, through: :signatures
```

Reply:
Does it make sense to keep it out of `Petition` just because it might never be used again after the running of this job?

I've gone with option B for now - definitely much more readable that what I had before 😁

Reply:
Yes, even though we main retain the functionality for future use - the stuff I wrote in 2017 for closing early was going to be removed because we'd want to think about it properly for the next election (which was going to be 2022 due to the Fixed Term Parliament Act) but then 2019 happened.